cmake.md

```diff
@@ -0,0 +1,30 @@
+## CMake build system
+### Terminology
+
+The CMake system in symbiflow-arch-defs uses the following hierarchy:
+
+* ARCH - The architecture of FPGA.  This is used to capture architecture level parameters (e.g. tech libraries).
+* DEVICE_TYPE - A type of device.  This is defined as an ARCH and the arch.xml which defines the units with this device type.
+* DEVICE - A specific instance of a DEVICE_TYPE.  A device definition is a DEVICE_TYPE, a PACKAGE list, and IO map definitions for each package if needed.
```

Comment:
Does `PACKAGE` == the layouts inside the arch.xml file?

Reply:
A package really pindefs, but the package is part of the --device during the rr_graph generation step.  The current CMake system assumes that each package needs its own rr_graph based on this argument.  So currently a package is a rr_graph and pindef map.

Reply:
Updated.